docs/ui-components.md

```diff
@@ -0,0 +1,168 @@
+# YouMe UI Components
+
+YouMe provides a number of onscreen components that are intended to control the [core grades](./core-components.md)
+provided by this package. These grades extend [`fluid.viewComponent`](https://docs.fluidproject.org/infusion/development/tutorial-gettingstartedwithinfusion/viewcomponents),
+which means that to instantiate them, you need to provide them with a selector that corresponds to a piece of existing
+markup. See the [demos](../demos) for examples.
+
+## `youme.templateRenderer`
+
+The UI components provided by YouMe make use of the "new view component" infrastructure in Infusion 4.x.  A key
+feature is that parent components render their own markup, and then associate child components with elements of the
+rendered markup.  Child components render themselves, and the process continues with grandchildren, et cetera.
+
+### Component Options
+
+| Name               | Type       | Description                                                                                                                                                        | Default       |
+|--------------------|------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------|
+| `markup.container` | `{String}` | A [string template](https://docs.fluidproject.org/infusion/development/coreapi#fluidstringtemplatetemplate-terms) representing the markup for this view component. | Empty string. |
+
+Please note, it is strongly recommended that you have [a single top-level element in your template](https://issues.fluidproject.org/browse/FLUID-6740).
+
+When rendering itself, the component will expose any material in its model for use as a variable.  So, if your component
+model contains a `label` variable, you might use a template like `<div class='my-label'>%label</div>` (see below).
+
+These templates do not keep track of updates to model material, you are expected to handle this yourself, as shown in
+the following example:
+
+```javascript
+var my = fluid.registerNamespace("my");
+
+fluid.defaults("my.widget", {
+    gradeNames: ["youme.templateRenderer"],
+    markup: {
+        container: "<div class='my-label'>%label</div>"
+    },
+    model: {
+        label: "Untitled"
+    },
+    selectors: {
+        label: ".my-label"
+    },
+    modelListeners: {
+        label: {
+            // Exclude "init" so that we don't interfere with the initial render.
```

Comment:
Reminder to self to make an "html" materialiser here. Is there any harm to missing off excludeSource: init? It seems that if the label isn't in the document yet the jQuery action will just be a noop?

Reply:
I think I was worried about duplicate render cycles, but everything works without it, so I got rid of them in the select box, multi select box, and these docs.